fili-core/src/test/groovy/com/yahoo/bard/webservice/data/config/table/BaseTableLoaderSpec.groovy

```diff
@@ -8,95 +8,269 @@ import com.yahoo.bard.webservice.data.config.ResourceDictionaries
+            this.dependentTableName = dependentTableName
+        }
+
+        SimpleDependencyPhysicalTableDefinition(String tableName, PhysicalTable physicalTable) {
+            super(TableName.of(tableName), DefaultTimeGrain.DAY.buildZonedTimeGrain(DateTimeZone.UTC), [] as Set, [] as Set)
+            this.physicalTable = physicalTable
+        }
+
+        @Override
+        Set<TableName> getDependentTableNames() {
+            if (Objects.isNull(dependentTableName)) {
+                return Collections.emptySet()
+            }
+            return Collections.singleton(TableName.of(dependentTableName));
+        }
+
+        @Override
+        Optional<PhysicalTable> build(
+                ResourceDictionaries dictionaries,
+                DataSourceMetadataService metadataService
+        ) {
+            if (!Objects.isNull(physicalTable)) {
+                return Optional.of(physicalTable)
+            }
+            return dictionaries.getPhysicalDictionary().containsKey(dependentTableName) ?
+                    Optional.of(
+                            new ConcretePhysicalTable(
+                                    TableName.of(getName().asName()),
+                                    DefaultTimeGrain.DAY.buildZonedTimeGrain(DateTimeZone.UTC),
+                                    [] as Set,
+                                    [:],
+                                    metadataService
+                            )
+                    ) : Optional.empty()
+        }
+    }
+
     BaseTableLoader loader
     ResourceDictionaries dicts
     Set<ApiMetricName> apiNames
     Set<FieldName> metricNames
     Set<PhysicalTableDefinition> physDefs
     Set<TestApiDimensionName> dimNames
     Collection<Dimension> dims
+    Set<TableName> tableNames
+    Set<PhysicalTableDefinition> allDependentDefinitions
+
+    PhysicalTableDefinition definition1
+    PhysicalTableDefinition definition2
+    PhysicalTableDefinition definition3
+    PhysicalTableDefinition definition4
+    PhysicalTableDefinition definition5
+    PhysicalTableDefinition definition6
+
+    PhysicalTable physicalTable
+    PhysicalTableSchema physicalTableSchema
 
     def setup() {
-        loader = new SimpleBaseTableLoader(new TestDataSourceMetadataService())
+        loader = new SimpleBaseTableLoader(Mock(DataSourceMetadataService))
         dicts = new ResourceDictionaries()
         apiNames = TestApiMetricName.getByLogicalTable(SHAPES)
         metricNames = TestDruidMetricName.getByLogicalTable(SHAPES)
-        physDefs = TestPhysicalTableDefinitionUtils.buildShapeTableDefinitions(new TestDimensions())
+        physDefs = TestPhysicalTableDefinitionUtils.buildShapeTableDefinitions(new TestDimensions(), metricNames)
         dimNames = TestApiDimensionName.getByLogicalTable(SHAPES)
+        tableNames = physDefs.stream().map({it -> it.getName()}).collect(Collectors.toSet())
```

Comment:
Since we're in groovy, it's generally more idiomatic to use groovy's built-in collection processing, closures, and property access:

```groovy
physDefs.collect {it.name} as Set
```